go/consensus/tendermint/abci/mux.go

```diff
@@ -568,7 +545,26 @@ func (mux *abciMux) BeginBlock(req types.RequestBeginBlock) types.ResponseBeginB
 	}
 
 	response := mux.BaseApplication.BeginBlock(req)
-	response.Events = ctx.GetEvents()
+
+	// Check any events left from InitChain.
+	_, evJSON := mux.state.deliverTxTree.Get([]byte(stateKeyInitChainEvents))
```

Comment:
Actually on second thought this will cause needless lookups on each block. Can we just look this up on on the first block?

Reply:
Fixed.